### PR TITLE
Showing warnings for old versions

### PR DESCRIPTION
**Pull request by souravgupta-msft**
Show security warnings to the user if they are using older version of blobfuse2.

### CODE REVIEW

**Review by vibhansa-msft — APPROVED** [review]
Looks good

**Comment by gapra-msft** [comment]
@souravgupta-msft can you merge main back into your branch? I messed up a little.

File: cmd/root.go [hunk 1]
```diff
@@ -112,13 +109,40 @@ func beginDetectNewVersion() chan interface{} {
...
+
+			_, isPresent := vJson.SecurityWarnings[common.Blobfuse2Version]
+			if isPresent {
+				hasWarning := false
+				ctr := 1
+				for _, msg := range vJson.SecurityWarnings[common.Blobfuse2Version] {
+					msg = strings.TrimSpace(msg)
+					if len(msg) > 0 {
+						if !hasWarning {
+							fmt.Fprintf(stderr, "Following vulnerabilities were detected in your current version (%s):\n", common.Blobfuse2Version)
```
Comment (gapra-msft) [inline cmd/root.go:123]: The following

Comment (gapra-msft) [reply cmd/root.go:123]: Also should we log these?

**Review by gapra-msft — APPROVED** [review]

**Review by tasherif-msft — APPROVED** [review]

File: cmd/root.go (hunk 1 above)
Comment (vibhansa-msft) [reply cmd/root.go:123]: I am not sure if logging makes sense here, its more of user needs to notice this and take action. Dumping to logs will go unnoticed. Also version check is happening before the logging lib is initialized.

Comment (souravgupta-msft) [reply cmd/root.go]: Added the warnings to log

**Pull request merged**